src/RequestHandler/managedexports.cxx

```diff
@@ -206,18 +206,26 @@ http_read_request_bytes(
     }
     IHttpRequest *pHttpRequest = (IHttpRequest*)pInProcessHandler->QueryHttpContext()->GetRequest();
 
-    BOOL fAsync = TRUE;
-
-    hr = pHttpRequest->ReadEntityBody(
-        pvBuffer,
-        dwCbBuffer,
-        fAsync,
-        pdwBytesReceived,
-        pfCompletionPending);
 
-    if (hr == HRESULT_FROM_WIN32(ERROR_HANDLE_EOF))
+    if (pHttpRequest->GetRemainingEntityBytes() > 0)
```

Comment:
How does this affect read performance? Is GetRemainingEntityBytes inexpensive when there is data available? Doing this on every read could be costly.

Reply:
Shouldn't affect read performance it returns 0 for GET requests and some value for Content-Lenght, for everything else `INFINITE` is returned.